doc/source/ray-core/examples/gentle_walkthrough.ipynb

```diff
@@ -0,0 +1,778 @@
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 9,
+   "id": "f5734bb1",
+   "metadata": {
+    "lines_to_next_cell": 2
+   },
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "((0, 'Learning'), (1, 'Ray'))\n",
+      "((2, 'Flexible'), (3, 'Distributed'))\n",
+      "((4, 'Python'), (5, 'for'))\n",
+      "((6, 'Machine'), (7, 'Learning'))\n"
+     ]
+    }
+   ],
+   "source": [
+    "@ray.remote\n",
+    "def follow_up_task(retrieve_result):\n",
+    "    original_item, _ = retrieve_result\n",
+    "    follow_up_result = retrieve(original_item + 1)\n",
+    "    return retrieve_result, follow_up_result\n",
+    "\n",
+    "\n",
+    "retrieve_refs = [retrieve_task.remote(item, db_object_ref) for item in [0, 2, 4, 6]]\n",
+    "follow_up_refs = [follow_up_task.remote(ref) for ref in retrieve_refs]\n",
+    "\n",
+    "result = [print(data) for data in ray.get(follow_up_refs)]"
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "id": "6e4a6945",
+   "metadata": {},
+   "source": [
+    "If you're unfamiliar with asynchronous programming, this example may not be particularly impressive.\n",
+    "However, at second glance it might be surprising that the code runs at all.\n",
+    "The code appears to be a regular Python function with a few list comprehensions.\n",
+    "\n",
+    "The function body of `follow_up_task` expects a Python tuple for its input argument `retrieve_result`.\n",
+    "However, when you use the `[follow_up_task.remote(ref) for ref in retrieve_refs]` command,\n",
+    "you are not passing tuples to the follow-up task.\n",
+    "Instead, you are using the `retrieve_refs` to pass in Ray object references.\n",
+    "\n",
+    "Behind the scenes, Ray recognizes that the `follow_up_task` needs actual values,\n",
+    "so it _automatically_ uses the `ray.get` function to resolve these futures.\n",
+    "Additionally, Ray creates a dependency graph for all the tasks and executes them in a way that respects their dependencies.\n",
+    "You don't have to explicitly tell Ray when to wait for a previous task to be completed––it infers the order of execution.\n",
+    "This feature of the Ray object store is useful because you avoid copying large intermediate values\n",
+    "back to the driver by passing the object references to the next task and letting Ray handle the rest."
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "id": "f8722673",
+   "metadata": {},
+   "source": [
+    "The next steps in the process are only scheduled once the tasks specifically designed to retrieve information are completed.\n",
+    "In fact, if `retrieve_refs` task was called something like `retrieve_result`, you might not have noticed this crucial and intentional naming nuance. Ray allows you to concentrate on your work rather than the technicalities of cluster computing.\n",
```

Comment:
```suggestion
    "In fact, if the `retrieve_refs` task was called something like `retrieve_result`, you might not have noticed this crucial and intentional naming nuance. Ray allows you to concentrate on your work rather than the technicalities of cluster computing.\n",
```